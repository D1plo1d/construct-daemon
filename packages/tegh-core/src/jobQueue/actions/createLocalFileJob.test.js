// TODO: move createLocalFileJob to a plugin or delete

it('needs to be refactored', () => {
  pending()
})
// import path from 'path'
// import tmp from 'tmp-promise'
//
// import fs from '../../util/promisifiedFS'
// import expectToMatchImmutableSnapshot from '../../util/testing/expectToMatchImmutableSnapshot'
// import createLocalFileJob from './createLocalFileJob'
//
// describe('createLocalFileJob', () => {
//   it('creates a CREATE_JOB action', async () => {
//     const tmpFile = await tmp.file({ postfix: '.gcode' })
//     const localPath = tmpFile.path
//
//     const getState = () => ({
//       config: {
//         printFromLocalPath: {
//           enabled: true,
//           whitelist: [
//             '/otherDirectory',
//             path.dirname(localPath),
//           ],
//         },
//       },
//     })
//     const dispatch = action => action
//
//     const result = await createLocalFileJob({
//       localPath,
//     })(dispatch, getState)
//
//     expectToMatchImmutableSnapshot({
//       result,
//       redactions: [
//         ['payload', 'job', 'name'],
//         ['payload', 'job', 'id'],
//         ['payload', 'job', 'createdAt'],
//         ['payload', 'jobFiles'],
//       ],
//     })
//
//     const jobFiles = Object.values(result.payload.jobFiles)
//     // eslint-disable-next-line no-restricted-syntax
//     for (const jobFile of jobFiles) {
//       expectToMatchImmutableSnapshot({
//         result: jobFile,
//         redactions: [
//           ['name'],
//           ['id'],
//           ['jobID'],
//           ['filePath'],
//         ],
//       })
//       // eslint-disable-next-line no-await-in-loop
//       const content = await fs.readFileAsync(jobFile.filePath, 'utf8')
//       expect(content).toMatchSnapshot()
//
//       /* clean up tmp file */
//       // eslint-disable-next-line no-await-in-loop
//       await fs.unlinkAsync(jobFile.filePath)
//     }
//   })
// })
