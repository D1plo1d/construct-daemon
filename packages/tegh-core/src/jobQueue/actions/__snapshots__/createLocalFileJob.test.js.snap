// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createLocalFileJob creates a CREATE_JOB action 1`] = `
Object {
  "payload": Object {
    "job": Object {
      "createdAt": "[REDACTED]",
      "id": "[REDACTED]",
      "name": "[REDACTED]",
      "quantity": 1,
    },
    "jobFiles": "[REDACTED]",
  },
  "type": "tegh/jobQueue/CREATE_JOB",
}
`;

exports[`createLocalFileJob creates a CREATE_JOB action 2`] = `
Object {
  "filePath": "[REDACTED]",
  "id": "[REDACTED]",
  "isTmpFile": false,
  "jobID": "[REDACTED]",
  "name": "[REDACTED]",
  "quantity": 1,
}
`;

exports[`createLocalFileJob creates a CREATE_JOB action 3`] = `""`;
